shorty.go

```diff
@@ -14,18 +14,49 @@ type Shorty struct {
 
 // New returns a new shorty, machineID can be any unique id to embed in the generated ids.
 func New(store Store, machineID uint32) *Shorty {
-	return &Shorty{
+	s := &Shorty{
 		s:         store,
 		machineID: machineID,
 	}
+	go s.cleanup()
+	return s
+}
+
+func (s *Shorty) cleanup() {
+	for {
+		var (
+			now     = time.Now()
+			expired []string
+		)
+		if s.s.ForEach(func(id string, v *Data) error {
+			if v.Expired(now) {
+				expired = append(expired, id)
+			}
+			return nil
+		}) != nil {
+			break
+		}
+		if len(expired) > 0 {
+			s.s.Delete(expired...)
+		}
+		time.Sleep(time.Hour)
+	}
 }
 
 // GenerateID generates a unique id for the provided url.
 func (s *Shorty) GenerateID(url string) (id ID, err error) {
+	return s.GenerateTimedID(url, 0)
```

Comment:
Well the reason I like -1 is because it's explicit. 0 could be an uninitialized value while -1 is very clearly set. @shahzilabid, tie breaker?